src/Sema.zig

```diff
@@ -27617,8 +27621,8 @@ fn explainWhyTypeIsNotPacked(
         .error_set,
         .@"anyframe",
         .optional,
-        .array,
         => try sema.errNote(src_loc, msg, "type has no guaranteed in-memory representation", .{}),
+        .array => try sema.errNote(src_loc, msg, "packed fields are ordered according to machine endianness, array elements are not", .{}),
```

Comment:
I'm not sure the error you gave is a good description of the problem; plus, it's kind of going a bit more into the weeds than is necessary. I'd suggest:
```suggestion
        .array => try sema.errNote(src_loc, msg, "array element ordering is incompatible with packed struct field ordering", .{}),
```

Reply:
I think explaining the precise reason here helps beginners to not hit a brick wall on this issue. How is it a bad description?
But if you think this is advanced enough that we should save the 9 characters, that's fine by me.

Reply:
It feels like packed structs field being ordered according to endianness is something we'd want to spell out in the langref, if anything. I don't think it actually says that explicitly.